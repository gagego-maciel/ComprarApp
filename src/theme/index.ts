export const theme = {
  colors: {
    primary: '#8257E5',
    background: '#121214',
    backgroundWhite: '#ffffff',
    backgroundSuccess: '#27ae60',
    backgroundButtonBlue: '#2c46b1',
    backgroundButtonRed: '#E53E3E',
    text: '#E1E1E6',
    textBlack: '#000000',
    textWhite: '#ffffff',
    textGray: '#A0A0A0',

    border: '#c3c5cb',

    placeHolderInput: '#74798B',
  },
}
