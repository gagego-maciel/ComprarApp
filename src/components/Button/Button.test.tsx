import React from 'react'
import { fireEvent, render } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { Button } from './index'
import { theme } from '@/theme'

const renderWithTheme = (component: React.ReactNode) =>
  render(<ThemeProvider theme={theme}>{component}</ThemeProvider>)

describe('Button Component', () => {
  test('should render correctly with text', () => {
    const { getByText } = renderWithTheme(
      <Button buttonText="Salvar" onPress={jest.fn()} />,
    )

    expect(getByText('Salvar')).toBeTruthy()
  })

  test('should call onPress when pressed', () => {
    const onPressMock = jest.fn()
    const { getByRole } = renderWithTheme(
      <Button buttonText="Pressionar" onPress={onPressMock} />,
    )

    fireEvent.press(getByRole('button'))
    expect(onPressMock).toHaveBeenCalledTimes(1)
  })

  test('should not call onPress when disabled', () => {
    const onPressMock = jest.fn()
    const { getByRole } = renderWithTheme(
      <Button buttonText="Desabilitado" onPress={onPressMock} disabled />,
    )

    fireEvent.press(getByRole('button'))
    expect(onPressMock).not.toHaveBeenCalled()
  })

  test('should apply disabled opacity style', () => {
    const { getByRole } = renderWithTheme(
      <Button buttonText="Disabled" onPress={jest.fn()} disabled />,
    )

    const button = getByRole('button')
    expect(button.props.style.opacity).toBe(0.5)
  })

  test('should match snapshott', () => {
    const tree = renderWithTheme(
      <Button buttonText="Snapshot" onPress={jest.fn()} />,
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
