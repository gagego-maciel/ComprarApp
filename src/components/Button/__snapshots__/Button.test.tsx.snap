// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshott 1`] = `
<View
  accessibilityRole="button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#2c46b1",
      "borderBottomLeftRadius": 8,
      "borderBottomRightRadius": 8,
      "borderTopLeftRadius": 8,
      "borderTopRightRadius": 8,
      "display": "flex",
      "height": 48,
      "justifyContent": "center",
      "opacity": 1,
      "paddingBottom": 0,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 0,
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#ffffff",
        "fontSize": 16,
        "fontWeight": "600",
      }
    }
  >
    Snapshot
  </Text>
</View>
`;
